pkgs/applications/window-managers/wayfire/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv, fetchFromGitHub
+, meson, pkgconfig, ninja
+, wayland, wayland-protocols
+, cairo, glm
+, libevdev, freetype, libinput
+, pixman, libxkbcommon, libdrm
+, libjpeg, libpng
+, libGL, mesa
+, libcap, xcbutilerrors, xcbutilwm, libxml2
+, buildDocs ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "wayfire";
+  version = "8aa8260972edaba6d237e1a61217a4d2303edc3e";
+
+  src = fetchFromGitHub {
+    owner = "WayfireWM";
+    repo = "wayfire";
+    rev = version;
+    sha256 = "sha256-r9k8e0LCDxr8tEZvcBbkvZ2RqcKXYX+/fxalKoqT15U=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ pkgconfig meson ninja ];
+  buildInputs = [
+    # egl glesv2
+    wayland wayland-protocols
+    cairo glm
+    libevdev freetype libinput
+    pixman libxkbcommon libdrm
+    libjpeg libpng
+    libGL mesa
+    libcap xcbutilerrors xcbutilwm libxml2
+  ];
+  mesonFlags = [
+    "-Duse_system_wlroots=disabled"
```

Comment:
We want to use the system libs.

Reply:
Can't the system version and the version wayfire uses aren't compatible fails to build the packages.